ouroboros-consensus/docs/ReleaseProcess.md

```diff
@@ -0,0 +1,229 @@
+# How Consensus Makes Releases
+
+This document explains how the Consensus team uses Pull Requests (PRs), version numbers, and branches to prepare releases of our packages.
+
+Let us assume our repository contains two packages FOO and BAR.
+Two is enough to explain the complexities.
+The resulting rules can be generalized to any number of packages, as explained at the end of the document.
+
+*Remark*.
+If you disagree with the decisions in this document, please consider reviewing our brainstorming notes and discussions, available at [this document's PR](https://github.com/input-output-hk/ouroboros-network/pull/4207).
+Even if you choose not to do so, please share your concern with us.
+
+## Notation
+
+In this document, we will assume each version number is a MAJOR.MINOR.PATCH triple.
+This is easy to generalize to other conventional notations, such as [the PVP](https://pvp.haskell.org/)'s MAJOR.MAJOR.MINOR.PATCH quadruples.
+
+We refer throughout this document to _the main branch_ (aka `master`, aka `main`).
+
+If you search for "RULE:" in this document, you'll find the key statements.
+
+## Rules for Branches
+
+The Consensus team will sometimes be maintaining multiple MAJOR.MINOR versions of the same package simultaneously.
+EG If we were currently maintaining a 2.3 version and a 2.4 version simultaneously, then we might release FOO-2.4.1 on Monday, release FOO-2.3.7 on Wednesday, and release FOO-2.4.2 on Friday, etc.
+
+- Whenever we're maintaining only the one greatest-ever MAJOR.MINOR version of a package, then all work on that package will happen on the main branch.
+- Whenever we're also maintaining some lesser MAJOR.MINOR versions of a package, then some work for that package will also be happening on the respective _release branches_.
+
+A _release branch_ is a branch dedicated to the maintenance of some older release of a package.
+
+RULE: If we have released a package FOO-X.Y.Z, then the release-FOO-X.Y.x branch MUST exist and its first-parent history MUST include the commit released as FOO-X.Y.Z.
```

Comment:
I am not opposed, but what would be a use case for backporting some change to a *proper* subset of minor versions of a given major release? IIUC, our users should always be able to upgrade from one minor versions to a later one under the same major version. If this extra flexibility is not required, we could just have one branch per major version.

Reply:
Hmm. I hadn't thought that through---I merely mimicked what I've seen in the `cardano-node` and `cardano-ledger` repositories.

Please confirm I'm understanding your point correctly. It should be as simple for the user to upgrade from A.B.X to A.(B+1).Y as it would be for them to upgrade to A.B.(C+1). And therefore we shouldn't bother maintain release branches for each MAJOR.MINOR pair, only one per MAJOR. Is that right?

In the abstract, that does seem right to me, because of the backwards-compatibility semantics of minor version bumps. But I wonder if there's some surprise lurking there that I'm failing to anticipate. ... maybe the downstream user has other constraints (eg regulatory/large company bullshit SoPs/etc) preventing them from taking the minor bump upgrade?

We can cross that bridge when/if we come to it, though. I'll adjust the text now.

Reply:
> Please confirm I'm understanding your point correctly. It should be as simple for the user to upgrade from A.B.X to A.(B+1).Y as it would be for them to upgrade to A.B.(X+1). And therefore we shouldn't bother maintaining release branches for each MAJOR.MINOR pair, only one per MAJOR. Is that right?

Exactly, thanks for spelling that out explicitly.